vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/reasoner/implementation/QueryAnswerSizeTest.java

```diff
@@ -0,0 +1,287 @@
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.Fact;
+import org.semanticweb.vlog4j.core.model.api.Literal;
+import org.semanticweb.vlog4j.core.model.api.PositiveLiteral;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+import org.semanticweb.vlog4j.core.model.implementation.Expressions;
+import org.semanticweb.vlog4j.core.reasoner.KnowledgeBase;
+
+public class QueryAnswerSizeTest {
+
+	private static final Predicate predP = Expressions.makePredicate("P", 1);
+	private static final Predicate predQ = Expressions.makePredicate("Q", 1);
+	private static final Predicate predR = Expressions.makePredicate("R", 2);
+	private static final Variable x = Expressions.makeUniversalVariable("x");
+	private static final Variable y = Expressions.makeExistentialVariable("y");
+	private static final Constant c = Expressions.makeAbstractConstant("c");
+	private static final Constant d = Expressions.makeAbstractConstant("d");
+	private static final Constant e = Expressions.makeAbstractConstant("e");
+	private static final Constant f = Expressions.makeAbstractConstant("f");
+
+	private static final PositiveLiteral Px = Expressions.makePositiveLiteral(predP, x);
+	private static final PositiveLiteral Qx = Expressions.makePositiveLiteral(predQ, x);
+	private static final PositiveLiteral Qy = Expressions.makePositiveLiteral(predQ, y);
+	private static final PositiveLiteral Rxx = Expressions.makePositiveLiteral(predR, x, x);
+	private static final PositiveLiteral Rxy = Expressions.makePositiveLiteral(predR, x, y);
+	private static final PositiveLiteral Ryy = Expressions.makePositiveLiteral(predR, y, y);
+
+	private static final Conjunction<PositiveLiteral> conRxyQy = Expressions.makePositiveConjunction(Rxy, Qy);
+	private static final Conjunction<PositiveLiteral> conRxxRxyRyy = Expressions.makePositiveConjunction(Rxx, Rxy, Ryy);
+	private static final Conjunction<Literal> conPx = Expressions.makeConjunction(Px);
+
+	private static final Rule QxPx = Expressions.makeRule(Qx, Px);
+	private static final Rule RxyQyPx = Expressions.makeRule(conRxyQy, conPx);
+	private static final Rule RxxRxyRyyPx = Expressions.makeRule(conRxxRxyRyy, conPx);
+
+	private static final Fact factPc = Expressions.makeFact(predP, c);
+	private static final Fact factPd = Expressions.makeFact(predP, d);
+
+	private static final Fact factQe = Expressions.makeFact(predQ, e);
+	private static final Fact factQf = Expressions.makeFact(predQ, f);
+
+	private static final PositiveLiteral Rdy = Expressions.makePositiveLiteral(predR, d, y);
+	private static final PositiveLiteral Rey = Expressions.makePositiveLiteral(predR, e, y);
+	private static final PositiveLiteral Rxd = Expressions.makePositiveLiteral(predR, x, d);
+	private static final PositiveLiteral Rxe = Expressions.makePositiveLiteral(predR, x, e);
+
```

Comment:
It should be tested that queries without variables return correct results.

For example,
 for a KB with fact ` f=P(c)`, `queryAnswerSize(f)` returns 1.